### ISSUE TITLE
Transformer should be able to read and write to distinct file systems

### ISSUE THREAD

**Issue by adamvinueza** [adamvinueza/transformer#25]
I might want to upload a local file to an S3 bucket, or transfer a file from an SFTP server to an S3 bucket. The file systems of the the source and destination in these cases have to be different.

Basic idea:
```
import s3fs
from fsspec.implementations import sftp

def decrypt(rdr, wr, params):
    """Decrypts the file read by rdr, writes decrypted stream using wr."""
    pass

src_fs = sftp.SFTPFileSystem(host, paramiko_params_dict)
dest_fs = s3fs.S3FileSystem(anon=False)
tr = Transform(src_fs=src_fs, dest_fs=dest_fs, overwrite=True)
tr('file_on_sftp_server.gpg', 's3://my_bucket/decrypted_file.csv', decrypt, [passphrase])
```

We should still be able to use a single file system, though:
```
import s3fs

def decrypt(rdr, wr, params):
    """Decrypts the file read by rdr, writes decrypted stream using wr."""
    pass

fs = s3fs.S3FileSystem(anon=False)
tr = Transform(fs=fs)
tr('s3://my_bucket/encrypted.gpg', 's3://my_bucket/decrypted.csv', decrypt, [passphrase])
```

Mixing the `fs` parameter with `src_fs` or `dest_fs` should raise an error; specifying only one of `src_fs` and `dest_fs` should also raise an error.

**Comment by adamvinueza** [comment]
Perhaps add methods to set the file systems?

```
import s3fs
from fsspec.implementations import sftp

tr = Transform()
tr.set_src_fs(sftp.SFTPFileSystem(host, paramiko_params_dict))
tr.set_dest_fs(s3fs.S3FileSystem(anon=False))
...

**Comment by adamvinueza** [comment]
Decided against having a single-file-system parameter--I don't yet see enough payoff in exchange for the additional complexity.

**Comment by adamvinueza** [comment]
Fixed in #26, version bumped in #27.
